addons/spreadsheet/static/tests/lists/list_plugin.test.js

```diff
@@ -417,6 +497,84 @@ test("can edit list domain", async () => {
     expect(getCellValue(model, "B2")).toBe("");
 });
 
+test("can edit list sorting", async () => {
+    const { model } = await createSpreadsheetWithList({
+        columns: ["foo", "bar", "date", "probability", "pognon"],
+    });
+    const [listId] = model.getters.getListIds();
+    expect(model.getters.getListDefinition(listId).orderBy).toEqual([]);
+    // prettier-ignore
+    expect(getEvaluatedGrid(model, "A1:E5")).toEqual([
+        ["Foo", "Bar",   "Date", "Probability", "Money!"],
+        [12,    "TRUE",  42474,  10,                74.4],
+        [1,     "TRUE",  42669,  11,                74.8],
+        [17,    "TRUE",  42719,  95,                   4],
+        [2,     "FALSE", 42715,  15,                1000],
+    ]);
+    const orderBy = [
+        {
+            name: "bar",
+            asc: false,
+        },
+        {
+            name: "pognon",
+            asc: true,
+        },
+    ];
+    const listDefinition = model.getters.getListModelDefinition(listId);
+    model.dispatch("UPDATE_ODOO_LIST", {
+        listId,
+        list: {
+            ...listDefinition,
+            searchParams: {
+                ...listDefinition.searchParams,
+                orderBy,
+            },
+        },
+    });
+    await waitForDataLoaded(model);
+    expect(model.getters.getListDefinition(listId).orderBy).toEqual(orderBy);
+    // prettier-ignore
+    expect(getEvaluatedGrid(model, "A1:E5")).toEqual([
+        ["Foo", "Bar",   "Date", "Probability", "Money!"],
+        [17,    "TRUE",  42719,   95,                  4],
+        [12,    "TRUE",  42474,   10,               74.4],
+        [1,     "TRUE",  42669,   11,               74.8],
+        [2,     "FALSE", 42715,   15,               1000],
+    ]);
+    undo(model);
+    expect(model.getters.getListDefinition(listId).orderBy).toEqual([]);
+    await waitForDataLoaded(model);
+    // prettier-ignore
+    expect(getEvaluatedGrid(model, "A1:E5")).toEqual([
+        ["Foo", "Bar",  "Date", "Probability", "Money!"],
```

Comment:
you can extract the 2 repeated grid in 2 variables `initialGrid` and `orderedGrid`